web/satellite/src/utils/plugins/segment.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+import loadScript from 'load-script';
+
+export class Segmentio {
+    analytics: any;
+    public init(key: string) {
+        if (this.analytics || key.length === 0 || key.includes('SegmentIOPublicKey')) {
+            return;
+        }
+        const source = `https://cdn.segment.com/analytics.js/v1/${key}/analytics.min.js`;
+        loadScript(source, (error) => {
+          if (error) {
+            return;
+          }
+
+          this.analytics = window['analytics'];
+        });
+
+        this.page();
+    }
+
+    public page() {
+        if (!this.analytics) {
+            return;
+        }
+
+        this.analytics.page();
+    }
+
+    public identify() {
+        if (!this.analytics) {
+            return;
+        }
+
+        this.analytics.identify();
+    }
+
+    public track(eventName: string, attributes: object) {
```

Comment:
and here we can specify event type like:
event: SegmentEvent
to prevent random string as parameter

Reply:
That's a good idea. I think @mbouzi can implement those when he goes to actually create all the events